### ISSUE TITLE
Fail on AttributeError when can not authenticate

### ISSUE THREAD

**Issue by lukas-bednar** [pycontribs/jira#298]
```
In [1]: from jira.client import JIRA

In [2]: j = JIRA(options=dict(server="https://issues.jboss.org", verify=True), basic_auth=("invalid_user", "invalid_password"), validate=True, max_retries=1)
WARNING:root:Bug https://jira.atlassian.com/browse/JRA-59676 trying again...
WARNING:root:Bug https://jira.atlassian.com/browse/JRA-59676 trying again...
WARNING:root:Bug https://jira.atlassian.com/browse/JRA-59676 trying again...
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-12-27f57a346db5> in <module>()
----> 1 j = JIRA(options=dict(server="https://issues.jboss.org", verify=True), basic_auth=("invalid_user", "invalid_password"), validate=True, max_retries=1)

/home/lbednar/work/jira/jira/client.pyc in __init__(self, server, options, basic_auth, oauth, jwt, kerberos, validate, get_server_info, async, logging, max_retries, proxies)
    305                 self._version = tuple(si['versionNumbers'])
    306             except Exception as e:
--> 307                 logging.error("invalid server_info: %s", si)
    308                 raise e
    309         else:

AttributeError: 'bool' object has no attribute 'error'
```

This issue was introduced by following commit:

https://github.com/pycontribs/jira/commit/97b36dfd1665e1693425aa264b08724afa4a9076#diff-12827fc5aab7f1c957381e453bdfe85aR301

The problem is that logging module is hidden by parameter passed to ``__init__``.

**Comment by eanxgeek** [comment]
I hit this bug yesterday, it seemed to just appear out of nowhere.  My program had been working and since I wrote it, over a week ago, but then yesterday this error started to surface.

I have python version 3.6.4 and the latest version of the Jira library:
$ python --version
Python 3.6.4 :: Anaconda custom (64-bit)
jira (1.0.10)                         - Python library for interacting with JIRA via REST
                                        APIs.
  INSTALLED: 1.0.10 (latest)

When I went into client.py I noticed this fix had not been applied, apparently a failed merge somewhere?  I applied the change and my program began working again.
